lib/internal/fs/utils.js

```diff
@@ -709,14 +709,30 @@ const validateOffsetLengthWrite = hideStackFrames(
   },
 );
 
-const validatePath = hideStackFrames((path, propName = 'path') => {
+const validatePath = hideStackFrames((path, propName = 'path', isFile) => {
```

Comment:
Can you add a JSDoc to this function and adding a small comment that explains the behavior of this new flag?

```suggestion
const validatePath = hideStackFrames((path, propName = 'path', expectFile) => {
```

Reply:
Fixed.